posts/2023-08-23-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,394 @@
+
+https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/Sso-realm/console/
+
+You should be able to log in successfully using the testuser1/testpasswd1.
+
+After logging in, in the General Section, you should see the endpoints. 
+
+Click on the link for the OpenID Endpoint Configuration - and that should point you to -
+https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+
+This will be needed for the client registration as the discoveryEndpoint later on.
+
+== Create the olapp-sso secret 
+
+Create a new project - called `gsm-test`, and create a secret in that namespace - - (Workloads->Secrets->Create Secret), called `guide-social-media-login-olapp-sso`, using key `oidc-clientId` and value `gsmapp`.
+
+The key name should be in this format <app-name>-olapp-sso. Use the same <app-name> as what you will use while deploying the applicaton using OLO.
+
+For e.g. in this setup, the <app-name> from the yaml file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`
+
+== Deploy the application to Open Shift using the Open Liberty Operator
+
+Since you have already installed the Open Liberty Operator, use the yaml file given below to deploy the Open Liberty App (guide-social-media-login) - for which you created the image using the Dockerfile mentioned above.
+
+Note the name of the application deployed is `guide-social-media-login`, the same name that was used when creating the secret above.
+
+Point to your image location in this section
+[source]
+----
+applicationImage: >-
+----
+and update the pullSecret needed to access the image here
+[source]
+----
+pullSecret: <secret_to_pull_image>
+----
+
+Also for the oidc: discoveryEndpoint - as shown here
+[source]
+----
+sso:
+    oidc:
+      - discoveryEndpoint: >-
+          https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+----
+
+point to the OpenID Endpoint Configuration that you configured while configuring the RH-SSO Operator.
+
+Here is the yaml file needed to deploy the application:
+
+[source]
+----
+apiVersion: apps.openliberty.io/v1
+kind: OpenLibertyApplication
+metadata:
+  name: guide-social-media-login
+  namespace: gsm-test
+spec:
+  sso:
+    oidc:
+      - discoveryEndpoint: >-
+          https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+  service:
+    port: 9443
+  applicationImage: >-
+    <image location of the app>
+  expose: true
+  manageTLS: true
+  replicas: 1
+  applicationName: guide-sm-login
+  pullPolicy: Always
+  pullSecret: <secret_to_pull_image>
+
+
+----
+== Create the OIDC Client in RH-SSO
+
+Since we have already deployed the guide-social-media-login app using the Open Liberty operator, we can now complete the registration for the openid client.
+
+Use this url - https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/master/console/
```

Comment:
Why? what does this URL allow us to access? Not clear, don't know why I'm doing this step.